### PR TITLE
Convert LicenseScanner to FossaApiClient

### PR DESCRIPTION
**Pull request by periodic**
# Overview

Converts `LicenceScanning` to use `FossaApiClient`.

Also, I had a structured called `ArchiveRevision` which I renamed to `PackageRevision`.  It captures the notion of an external dependency that will never have a branch and so `ProjectRevision` is not appropriate.  I'm not sure if this is the best naming, but it makes the API clearer to bundle these instead of passing around pairs of `Text` values.  I welcome discussion on this.

I also noticed there were two places where we were ignoring the value of the call, so I converted those to just return unit.

## Acceptance criteria

- All of `LicenceScanning` uses `FossaApiClient`.
- Tests added where feasible

## Testing plan

The function exposed by `LicenseScanner` actually executes Themis.  Our test harnesses don't seem to have mock execution set up.  This makes testing hard because I'd need to create and run against a fake dependency.  That feels out of scope for this PR, so I didn't create any unit tests.

I tested by incrementally implementing the endpoints to ensure everything was working.

1. First, set up my local core to do local license scanning.
2. Run the CLI with local scanning against a vendored dep and observe the behavior, verifying it invoked local scanning.
3. Refactor the code up to the point that it would invoke the legacy API functions. 
 Those are replaced with dummy implementations using `fatalText`.
4. Run the CLI and observe the dummy errors, updating the triggered dummy implementation with a pass-through to the API V1 implementation.
5. Repeat until license scanning succeeds.
6. Double-check that all expected end-points were invoked to ensure I have full coverage.

Future testing is just to run a local licence scan and observe that it works.

## References

- [ANE-217](https://fossa.atlassian.net/browse/ANE-217): Convert License Scanning to use FossaApiClient

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

### CODE REVIEW

File: src/App/Fossa/LicenseScanner.hs [hunk 1]
```diff
@@ -12,16 +12,15 @@ import App.Fossa.ArchiveUploader (
...
  )
 import App.Fossa.EmbeddedBinary (ThemisBins, withThemisAndIndex)
-import App.Fossa.FossaAPIV1 qualified as Fossa
 import App.Fossa.RunThemis (
   execThemis,
  )
 import Control.Carrier.Finally (runFinally)
 import Control.Effect.Diagnostics (Diagnostics, ToDiagnostic (renderDiagnostic), context, fatal, fatalText, fromMaybe, recover)
-import Control.Effect.FossaApiClient (FossaApiClient, getApiOpts)
+import Control.Effect.FossaApiClient (FossaApiClient, PackageRevision (PackageRevision, packageName, packageVersion), finalizeLicenseScan, getOrganization, getSignedLicenseScanUrl, uploadLicenseScanResult)
```
Comment (meghfossa) [inline src/App/Fossa/LicenseScanner.hs:20]: [optional] [nit] line break in the listing for better formatting

File: src/App/Fossa/ArchiveUploader.hs [hunk 2]
```diff
@@ -15,7 +15,7 @@ import Codec.Compression.GZip qualified as GZip
 import Control.Carrier.Diagnostics qualified as Diag
 import Control.Carrier.StickyLogger (StickyLogger, logSticky)
 import Control.Effect.Diagnostics (context)
-import Control.Effect.FossaApiClient (ArchiveRevision (ArchiveRevision), FossaApiClient, getOrganization, getSignedUploadUrl, queueArchiveBuild, uploadArchive)
+import Control.Effect.FossaApiClient (FossaApiClient, PackageRevision (PackageRevision), getOrganization, getSignedUploadUrl, queueArchiveBuild, uploadArchive)
```
Comment (meghfossa) [inline src/App/Fossa/ArchiveUploader.hs:18]: [optional] [nit] line break in the listing for better formatting

Comment (meghfossa) [inline src/App/Fossa/ArchiveUploader.hs:18]: [optional] [nit] line break in the listing for better formatting

File: src/App/Fossa/LicenseScanner.hs [hunk 3]
```diff
@@ -216,10 +213,10 @@ uploadVendoredDep apiOpts VendoredDependency{..} themisScanResult = do
     Nothing -> pure . showT . md5 . encodeUtf8 . show . NE.sort $ themisScanResult
     Just version -> pure version
 
-  signedURL <- Fossa.getSignedLicenseScanURL apiOpts depVersion vendoredName
+  signedURL <- getSignedLicenseScanUrl $ PackageRevision{packageVersion = depVersion, packageName = vendoredName}
```
Comment (meghfossa) [inline src/App/Fossa/LicenseScanner.hs:216]: [optional] [nit] why not `PackageRevision vendoredName depVersion` like you have in the Archive uploader?

Comment (meghfossa) [inline src/App/Fossa/LicenseScanner.hs:216]: [optional] [nit] why not `PackageRevision vendoredName depVersion` like you have in the Archive uploader?

**Review by meghfossa — APPROVED** [review]
left some options. Did not do e/e test but the changes look straightforward.

File: src/App/Fossa/LicenseScanner.hs (hunk 3 above)
Comment (periodic) [reply src/App/Fossa/LicenseScanner.hs:216]: I actually think they should both be this way.  The point is to name these `Text` parameters that are being passed in so it's hard to accidentally get them switched around.

Comment (periodic) [reply src/App/Fossa/LicenseScanner.hs:216]: Looks like GitHub posted some duplicates.

**Pull request merged**